### ISSUE TITLE
Bug: System.Security.Permissions Dependency Mismatch with CMSApp (NU1605)

### ISSUE THREAD

**Issue by seangwright** [Kentico/xperience-algolia#23]
### Brief bug description

The `CMSApp` project has a dependency on `System.Security.Permissions` (listed in `packages.config`)

```xml
<package id="System.Security.Permissions" version="4.5.0" targetFramework="net48" />
```

This library has a dependency on `System.Configuration.ConfigurationManager`, which has a [transitive dependency](https://www.nuget.org/packages/System.Configuration.ConfigurationManager/6.0.0#dependencies-body-tab) on `System.Security.Permissions` >= 6.0.0

By default this can result in warnings with .NET restore/build since these versions are incompatible.

### Repro steps

- Install the `Kentico.Xperience.AlgoliaSearch` library in a project that `CMSApp` takes a dependency on (or in `CMSApp` directly).
- Perform a `dotnet restore`
- See the warning in the terminal - `warning NU1605: Detected package downgrade:`

### Expected behavior

Versions of transitive dependencies are compatible between this library and the .NET 4.8 `CMSApp` project

What is the correct behavior?

Either upgrade `System.Security.Permissions` in `CMSApp` to 6.0.x or downgrade the dependency in this library to `4.5.0`.

Is `CMSApp` compatible with `System.Security.Permissions` `6.0.0`? If so, that resolves the issue, but there's nothing documented about it in this library or on [docs.xperience.io](https://docs.xperience.io/installation/hotfix-instructions-xperience-13)

### Test environment

.NET SDK 6.0.301

### Additional context

`System.Security.Permissions` [supports net48](https://www.nuget.org/packages/System.Security.Permissions/#supportedframeworks-body-tab), so updating the `CMSApp` dependency on `System.Security.Permissions` seems like the best guidance, assuming it's supported by the logic/API calls in `CMSApp`.

### Screenshots

n/a

**Comment by kentico-ericd** [comment]
I wasn't able to reproduce this behavior on a fresh installation of Xperience Refresh 6. When I installed the `Kentico.Xperience.AlgoliaSearch` package, the `System.Security.Permission` was upgraded to 6.0.0 and the older version was removed:

>Adding package 'System.Security.Permissions.6.0.0' to folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Added package 'System.Security.Permissions.6.0.0' to folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Added package 'System.Security.Permissions.6.0.0' to 'packages.config'
Successfully installed 'System.Security.Permissions 6.0.0' to CMSApp
Adding package 'System.Configuration.ConfigurationManager.6.0.0' to folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Added package 'System.Configuration.ConfigurationManager.6.0.0' to folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Added package 'System.Configuration.ConfigurationManager.6.0.0' to 'packages.config'
Successfully installed 'System.Configuration.ConfigurationManager 6.0.0' to CMSApp
...
Successfully installed 'Kentico.Xperience.AlgoliaSearch 2.3.0' to CMSApp
Removing package 'System.Security.AccessControl.4.5.0' from folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Removed package 'System.Security.AccessControl.4.5.0' from folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Removing package 'System.Security.Permissions.4.5.0' from folder 'C:\inetpub\wwwroot\TestInstallation\packages'
Removed package 'System.Security.Permissions.4.5.0' from folder 'C:\inetpub\wwwroot\TestInstallation\packages'

We haven't encountered any issues using v6.0.0 while testing, so it should be okay. Am I missing some step to reproduce the issue?
